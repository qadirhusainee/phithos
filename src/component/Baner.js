import React, { useState, useEffect } from "react";
import { useSelector } from "react-redux";
import type { FunctionComponent } from "react";
import {
  View,
  Text,
  StyleSheet,
  Button,
  TextInput,
  Dimensions,
  Platform,
} from "react-native";
import { isEmulator } from "react-native-device-info";
const { width, height } = Dimensions.get("window");

const CustomBaner: FunctionComponent = ({ navigation }) => {
  const showBaner = useSelector((state) => state.auth.showBaner);
  const [isDevice, setIsDevice] = useState("");
  const [banerVisible, setBanerVisible] = useState(true);

  useEffect(() => {
    getDeviceInfo();
  }, []);

  const getDeviceInfo = async () => {
    const response = await isEmulator();
    setIsDevice(!response);
  };

  if (showBaner && banerVisible) {
    return (
      <View style={styles.container}>
        <Text style={{ fontSize: 20, paddingTop: 5 }}>
          {isDevice
            ? "Device"
            : Platform.OS == "ios"
            ? "Simulator(IOS)"
            : "Emulator(Android)"}
        </Text>
        <View
          style={{
            width: width,
            justifyContent: "flex-end",
            flexDirection: "row",
            marginRight: 20,
          }}
        >
          <Button onPress={() => setBanerVisible(false)} title="Close" />
        </View>
      </View>
    );
  } else {
    return <View />;
  }
};

const styles = StyleSheet.create({
  container: {
    width: width,
    alignItems: "center",
    justifyContent: "center",
    height: 80,
    borderWidth: 0.5,
    borderColor: "grey",
    backgroundColor: "#f2f2f2",
  },

  error: {
    color: "red",
  },
  button: {
    marginTop: 20,
  },
});

export default CustomBaner;
