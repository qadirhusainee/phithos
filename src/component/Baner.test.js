import React from "react";
import { render } from "react-native-testing-library";
import renderer from "react-test-renderer";

import Baner from "./Baner";

jest.mock("react-redux", () => {
  const ActualReactRedux = require.requireActual("react-redux");
  return {
    ...ActualReactRedux,
    useSelector: jest.fn().mockImplementation(() => {
      return { showBaner: true };
    }),
  };
});
describe("Baner", () => {
  it("Matches Snapshot", async () => {
    const tree = await renderer.create(<Baner />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Showing device or emulator", async () => {
    const { queryByText } = render(<Baner />);

    const element = queryByText("Emulator(Android)");
    expect(element).toBeDefined();
  });
});
