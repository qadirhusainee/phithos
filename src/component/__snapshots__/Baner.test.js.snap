// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Baner Matches Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f2f2f2",
      "borderColor": "grey",
      "borderWidth": 0.5,
      "height": 80,
      "justifyContent": "center",
      "width": 750,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "paddingTop": 5,
      }
    }
  >
    Simulator(IOS)
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "marginRight": 20,
        "width": 750,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Close
        </Text>
      </View>
    </View>
  </View>
</View>
`;
